renku/cli/_providers/dataverse.py

```diff
@@ -0,0 +1,254 @@
+
+        uri = self.get_export_uri(uri)
+
+        return self.get_record(uri)
+
+    def get_export_uri(self, uri):
+        """Gets a dataverse api export URI from a dataverse entry."""
+        record_id = DataverseProvider.record_id(uri)
+        uri = make_records_url(record_id, uri)
+        return uri
+
+    def get_record(self, uri):
+        """Retrieve metadata and return ``DataverseRecordSerializer``."""
+        response = self.make_request(uri)
+
+        return DataverseRecordSerializer(
+            json=response.json(), dataverse=self, uri=uri
+        )
+
+    def get_exporter(self, dataset, access_token):
+        """Create export manager for given dataset."""
+        raise NotImplementedError()
+
+
+@attr.s
+class DataverseRecordSerializer:
+    """Dataverse record Serializer."""
+
+    _dataverse = attr.ib(default=None, kw_only=True)
+
+    _uri = attr.ib(default=None, kw_only=True)
+
+    _json = attr.ib(default=None, kw_only=True)
+
+    def is_last_version(self, uri):
+        """Check if record is at last possible version."""
+        return True
+
+    @property
+    def files(self):
+        """Get all file metadata entries."""
+        file_list = []
+
+        for f in self._json['distribution']:
+            mapped_file = {k.strip('@'): v for k, v in f.items()}
+            mapped_file['parentUrl'] = self._uri
+            file_list.append(mapped_file)
+        return file_list
+
+    def get_jsonld(self):
+        """Get record metadata as jsonld."""
+        response = self._dataverse.accept_jsonld().make_request(self._uri)
+        self._jsonld = response.json()
+        return self._jsonld
+
+    def get_files(self):
+        """Get Dataverse files metadata as ``DataverseFileSerializer``."""
+        if len(self.files) == 0:
+            raise LookupError('no files have been found')
+
+        return [DataverseFileSerializer(**file_) for file_ in self.files]
+
+    def as_dataset(self, client):
+        """Deserialize `DataverseRecordSerializer` to `Dataset`."""
+        files = self.get_files()
+        dataset = Dataset.from_jsonld(self._json, client=client)
+
+        serialized_files = []
+        for file_ in files:
+            remote_ = file_.remote_url
+            dataset_file = DatasetFile(
+                url=remote_,
+                id=file_._id if file_._id else file_.name,
+                filename=file_.name,
+                filesize=file_.contentSize,
+                filetype=file_.fileFormat,
+                dataset=dataset.name,
+                path='',
+            )
+            serialized_files.append(dataset_file)
+
+        dataset.files = serialized_files
+
+        return dataset
+
+
+@attr.s
+class DataverseFileSerializer:
+    """Dataverse record file."""
+
+    _id = attr.ib(default=None, kw_only=True)
+
+    identifier = attr.ib(default=None, kw_only=True)
+
+    name = attr.ib(default=None, kw_only=True)
+
+    fileFormat = attr.ib(default=None, kw_only=True)
```

Comment:
I see it's convenient, but ideally we want to keep same notation through out the code base. Can we rename this to `file_format`?